src/webloader.js

```diff
@@ -0,0 +1,106 @@
+import { generate } from './guids.js';
+import EvolvAssetManager from './index.js';
+import EvolvClient from "@evolv/javascript-sdk";
+
+
+function ensureId(evolv, key, session) {
+	let id;
+	if (evolv.context && key in evolv.context) {
+		id = evolv[key];
+	} else {
+		id = evolv.retrieve(key, session);
+		if (!id) {
+			id = generate();
+			evolv.store(key, id, session);
+		}
+	}
+	return id;
+}
+
+function currentScript() {
+	if (document.currentScript) {
+		return document.currentScript;
+	}
+
+	for (let i = 0; i < document.scripts.length; i++) {
+		if ('env' in document.scripts.dataset) {
+			return document.scripts;
+		}
+	}
+}
+
+function injectScript(endpoint, env, uid) {
+	const script = document.createElement('script');
+	script.type = 'text/javascript';
+	script.src = endpoint + '/' + env + '/' + uid + '/assets.js';
+
+	document.head.appendChild(script);
+}
+
+function injectStylesheet(endpoint, env, uid) {
+	const stylesheet = document.createElement('link');
+	stylesheet.setAttribute('rel', 'stylesheet');
+	stylesheet.setAttribute('type', 'text/css');
+	stylesheet.setAttribute('href', endpoint + '/' + env + '/' + uid + '/assets.css');
+
+	document.head.appendChild(stylesheet);
+}
+
+if (!window.evolv) {
```

Comment:
I had a question about where we should do this. When i was using the document.getElementsByTagName() api and the asset.js was lower than the AssetManager on the page, the api could not find the script. So unless the other apis (document.scripts) can find a lower script it almost seems like we need to require that asset.js be higher than the AssetManager. At that point should the asset.js set up the evolv object on the window?